dev-utils/src/commands/init_command.rs

```diff
@@ -0,0 +1,308 @@
+            .map(|mut file| file.write_all(json_string.as_bytes()))
+            .map(|_e| {
+                println!(
+                    "writing tendermint genesis OK {}",
+                    self.get_tendermint_filename()
+                );
+            })
+            .map_err(|_e| format_err!("write tendermint genesis error"))
+    }
+    fn prepare_tendermint(&self) -> Result<(), Error> {
+        // check whether file exists
+        fs::read_to_string(&self.get_tendermint_filename())
+            .or_else(|_e| {
+                // file not exist
+                Command::new("tendermint")
+                    .args(&["init"])
+                    .output()
+                    .map(|_e| {
+                        println!("tenermint initialized");
+                        "".to_string()
+                    })
+                    .map_err(|_e| format_err!("tendermint not found"))
+            })
+            .map(|_e| ())
+    }
+    fn read_staking_address(&mut self) -> Result<(), Error> {
+        let storage = SledStorage::new(InitCommand::storage_path())?;
+        let wallet_client = DefaultWalletClient::builder()
+            .with_wallet(storage)
+            .build()?;
+
+        let name = input()
+            .msg("please enter wallet name=")
+            .default("my".to_string())
+            .get();
+        let passphrase = InitCommand::ask_passphrase()?;
+        match wallet_client.new_wallet(&name.as_str(), &passphrase) {
+            Ok(_a) => {}
+            Err(b) => {
+                println!("new wallet fail={}", b.to_string());
+            }
+        }
+        success(&format!("Wallet created with name: {}", name));
+
+        let address = wallet_client.new_staking_address(&name.as_str(), &passphrase)?;
+        success(&format!("New address: {}", address));
+        self.staking_account_address = address.to_string().trim().to_string();
+        println!("staking address={}", self.staking_account_address);
+        assert!(address.to_string().trim().to_string().len() == 42);
+        Ok(())
+    }
+    pub fn execute(&mut self) -> Result<(), Error> {
+        println!("initialize");
+        self.prepare_tendermint()
+            .and_then(|_| self.read_tendermint_genesis())
+            .and_then(|_| self.read_staking_address())
+            .and_then(|_| self.read_information())
+            .and_then(|_| self.generate_app_info())
+            .and_then(|_| self.write_tendermint_genesis())
+            .map_err(|e| format_err!("init error={}", e))
+    }
+
+    fn storage_path() -> String {
+        match std::env::var("CRYPTO_CLIENT_STORAGE") {
+            Ok(path) => path,
+            Err(_) => ".storage".to_owned(),
+        }
+    }
+
+    fn ask_passphrase() -> client_common::Result<SecUtf8> {
+        InitCommand::ask("Enter passphrase: ");
+        Ok(password().context(ErrorKind::IoError)?.into())
+    }
+
+    /// Print a question, in bold, without creating a new line.
+    fn ask(q: &str) {
```

Comment:
`quest` crate already has these utility functions built-in. I think its better to use those instead of creating our own. The ones in `quest` are also tested against different OSes.

Reply:
agree, i will change it